src/mixer.c

```diff
@@ -635,8 +635,8 @@ static SDL_AudioSpec *Mix_LoadMusic_RW(SDL_RWops *src, int freesrc, SDL_AudioSpe
 
     Mix_UnlockAudio();
 
-    if (count > 0) {
-        *audio_len = (count - 1) * fragment_size + fragment->size;
+    if (count > 0 && last) {
```

Comment:
Aren't we guaranteed that last will not be NULL if count > 0?

Reply:
> Aren't we guaranteed that last will not be NULL if count > 0?

I guess, yes, you're right, then, the "last" check can be easily removed.

Reply:
Made

Reply:
Perfect! :)